stdlib/src/main/scala/chisel3/std/tilelink/Parameters.scala

```diff
@@ -0,0 +1,58 @@
+package chisel3.std.tilelink
+
+case class TLBundleParameter(
+  a: TLChannelAParameter,
+  b: Option[TLChannelBParameter],
+  c: Option[TLChannelCParameter],
```

Comment:
Splitting the parameters across by channels might be dangerous.
What happens if a.addressBits != b.addressBits, for example?

What is the use case for this?

Reply:
Yes, I do think so. However the TileLink Spec doesn't provide any constraint on this, I also think there two parameter should be same. But that should be added to TileLink spec instead...

I think we need to add a general sanity check library to check these constraint. and busip can use the library directly.

Reply:
I see. If the underlying implmentation breaks it into parameters by channel, there should still be a constructor for creating a "unified" bundle, consisting of all 5 channels, similar to the TLBundleParameters in rocket-chip.

Reply:
Now there is two methods `UncachedBus` and `CachedBus` in the companion object for creating such a "unified" bundle. And the test is updated to use them.

Reply:
What is the benefit of splitting them into two separate constructors, instead of just having `supportsBCE` as another parameter (as rocket-chip does).

This approach yields to some ugliness with
```
if (cached) 
  TLBundleParameter.Cached(addrBits, sourceBits, ...) 
else 
  TLBundleParameter.Uncached(addrBits, sourceBits, ...)
```

Reply:
I thought most bus peripherals would only use one of TL-UL, TL-UH or TL-C levels exclusively. On a crossbar a dynamic parameter can make sense, though.

Reply:
What about things like buffers, or clock crossings? I think the `hasBCE` approach is much neater,